### PR TITLE
Feature/matrix 1d 2d edit

### LINKED ISSUE #1940: Matrix1D Compact Edit and View Input
### New Addition
* Add Matrix1D compact edit form

* Add readonly mode

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP_UI_LIBRARY-%5Bin-progress%5D?node-id=6887%3A6581&scaling=min-zoom)

>NOTE: Need further investigation @tnagorra @AdityaKhatri

### LINKED ISSUE #1941: Matrix2D compact edit and view input
### New Addition
* Add Matrix2D compact edit form

* Add readonly mode

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP_UI_LIBRARY-%5Bin-progress%5D?node-id=6887%3A6581&scaling=min-zoom)

>NOTE: Need further investigation @tnagorra @AdityaKhatri

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Add list widget input and readonly for matrix1D and matrix2D

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/AnalyticalFramework/ListWidgetPreview/index.tsx
Comment (tnagorra) [inline app/views/AnalyticalFramework/ListWidgetPreview/index.tsx]: Oh no, we are going to have a very nice conflict

File: app/utils/common.ts
Comment (tnagorra) [inline app/utils/common.ts]: Looks like this is not used!

File: app/views/AnalyticalFramework/ListWidgetPreview/Matrix1dWidgetInput/index.tsx
Comment (tnagorra) [inline app/views/AnalyticalFramework/ListWidgetPreview/Matrix1dWidgetInput/index.tsx]: The data has been renamed to properties

**Review by tnagorra — COMMENTED** [review]
We may need to wait for the other PR to merge

File: app/views/AnalyticalFramework/ListWidgetPreview/index.tsx
Comment (AdityaKhatri) [reply app/views/AnalyticalFramework/ListWidgetPreview/index.tsx]: haha yes

File: app/utils/common.ts
Comment (AdityaKhatri) [reply app/utils/common.ts]: yes its not :dagger:

File: app/views/AnalyticalFramework/ListWidgetPreview/Matrix1dWidgetInput/index.tsx
Comment (AdityaKhatri) [reply app/views/AnalyticalFramework/ListWidgetPreview/Matrix1dWidgetInput/index.tsx]: will fix it after merge

File: app/views/AnalyticalFramework/FrameworkForm/components/ListCanvas/ListWidgetPreview/Matrix1dWidgetInput/index.tsx
Comment (tnagorra) [inline app/views/AnalyticalFramework/FrameworkForm/components/ListCanvas/ListWidgetPreview/Matrix1dWidgetInput/index.tsx]: Let's not use default value

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**
